File: mobile-monorepo/packages/common/lib/src/utils/auth_utils.dart
import '../model/dto/token_dto.dart';

bool isTokenExpired(
  TokenResponse? token,
  DateTime? issuedAt,
) {
  if (token == null || issuedAt == null || token.accessTokenExpiresIn <= 0) {
    return true;
  }

  final expirationTime = issuedAt.add(
    Duration(seconds: token.accessTokenExpiresIn),
  );
  return !DateTime.now().isBefore(expirationTime);
}

bool isTokenExpiringSoon(
  TokenResponse? token,
  DateTime? issuedAt, {
  int minutesBuffer = 5,
}) {
  if (token == null || issuedAt == null || token.accessTokenExpiresIn <= 0) {
    return true;
  }

  final expirationTime = issuedAt.add(
    Duration(seconds: token.accessTokenExpiresIn),
  );
  final bufferTime = Duration(minutes: minutesBuffer);
  return !DateTime.now().isBefore(expirationTime.subtract(bufferTime));
}

File: turborepo/packages/common/utils/auth-utils.ts
import { TokenDto } from '../model/dto/token-dto'

export const isTokenExpired = (token: TokenDto.TokenResponse | null, issuedAt: number | null): boolean => {
  if (!token?.accessTokenExpiresIn || issuedAt === null) {
    return true
  }

  const now = Date.now()
  const expirationTime = issuedAt + (token.accessTokenExpiresIn * 1000)

  return now >= expirationTime
}

export const isTokenExpiringSoon = (token: TokenDto.TokenResponse | null, issuedAt: number | null, minutesBuffer = 5): boolean => {
  if (!token?.accessTokenExpiresIn || issuedAt === null) {
    return true
  }

  const now = Date.now()
  const expirationTime = issuedAt + (token.accessTokenExpiresIn * 1000)
  const bufferTime = minutesBuffer * 60 * 1000

  return now >= (expirationTime - bufferTime)
}

